src/routes/news/containers/NewsPage.js

```diff
@@ -12,80 +12,57 @@ import type { StateType } from '../../../modules/app/StateType'
 import NewsElement from '../components/NewsElement'
 import NewsList from '../components/NewsList'
 import Tabs from '../components/Tabs'
-import LocalNewsDetails from '../components/LocalNewsDetails'
+import LocalNewsDetailsPage from './../containers/LocalNewsDetails'
 import TuNewsDetails from './TuNewsDetails'
 import TuNewsElement from '../components/TuNewsElement'
+import Link from 'redux-first-router-link'
+import { redirect } from 'redux-first-router'
+import type { Dispatch } from 'redux'
+import type { StoreActionType } from '../../../modules/app/StoreActionType'
+import { TUNEWS_LIST_ROUTE } from './../../../modules/app/route-configs/TuNewsListRouteConfig'
+import { CATEGORIES_ROUTE } from './../../../modules/app/route-configs/CategoriesRouteConfig'
 
 type PropsType = {|
   news: Array<LocalNewsModel>,
-  tuNews: Array<TuNewsElementModel>,
   city: string,
-  newsId: ?string,
   language: string,
   t: TFunction,
-  path: string
+  path: string,
+  cities: Array<CityModel>,
+  redirect: any,
 |}
 
-const LOCAL_NEWS = 'local'
-const TU_NEWS = 'tu'
-
 /**
  * Displays a list of news or a single newsElement, matching the route /<location>/<language>/news(/<id>)
  */
 export class NewsPage extends React.Component<PropsType> {
-  state: any = {
-    hasNextPage: true,
-    isNextPageLoading: false,
-    items: []
-  };
 
-  renderLocalNewsElement = (language: string, type: string) => (newsItem: LocalNewsModel, city: string) => {
-    return (
-      <NewsElement
-        newsItem={newsItem}
-        key={newsItem.path}
-        path={this.props.path}
-        t={this.props.t}
-        language={language}
-      />
-    )
-  }
+  componentDidMount() {
 
-  render() {
-    const { news, path, newsId, city, language, t, tuNews } = this.props
-    if (newsId) {
-      const newsItem = news.find(_newsItem => {
-        return _newsItem.title === decodeURIComponent(newsId)
-      })
+    const currentCity: any = this.props.cities.find(cityElement => cityElement._code === this.props.city) || {}
 
-      if (newsItem) {
-        const { title, message, timestamp } = newsItem
-        return (
-          <>
-            <LocalNewsDetails title={title} message={message} timestamp={timestamp} language={language} />
-          </>
-        )
-      } else {
-        const error = new ContentNotFoundError({ type: 'newsItem', id: newsId, city, language })
-        return <FailureSwitcher error={error} />
-      }
+    if (!currentCity.newsEnabled && !currentCity.tuNewsEnabled) {
```

Comment:
@orajaby Please move this logic to separate method, maybe put it in shared class or something like that since its used in many places